gef.py

```diff
@@ -950,16 +952,16 @@ class Shdr:
     SHF_ORDERED          = 0x40000000
     SHF_EXCLUDE          = 0x80000000
 
-    sh_name      = None
-    sh_type      = None
-    sh_flags     = None
-    sh_addr      = None
-    sh_offset    = None
-    sh_size      = None
-    sh_link      = None
-    sh_info      = None
-    sh_addralign = None
-    sh_entsize   = None
+    sh_name: Optional[int]      = None
```

Comment:
Those are all int except `sh_name`, might as well initialize them directly (0 -> int, "" -> str)
If so, L991 can be removed too

Reply:
I was confused about the following lines for the type of `sh_name`

https://github.com/hugsy/gef/blob/9560041891177335de88b3b2d5cf3ce30db64fdd/gef.py#L971
https://github.com/hugsy/gef/blob/9560041891177335de88b3b2d5cf3ce30db64fdd/gef.py#L976
https://github.com/hugsy/gef/blob/9560041891177335de88b3b2d5cf3ce30db64fdd/gef.py#L989

So I guess `sh_name` is first `int` (offset into file where the name is located) and then the same variable is used later for the actual `str` value. So the type is `Union[int,str]` with an inital value of `0` and L991 **can't** be removed

Reply:
@hugsy let me know if you agree with my latest commited version for this

Reply:
You're right, I forgot `sh_name` is an offset (despite its name). Which means that here in our implementation, we lose the offset value when we get the string. So new suggestion (roughtly):

```python
def __init__():
  [...]
  sh_name: int  = 0
  [...]

@property
def name(self):
        while True:
            c = ord(elf.read(1))
            if c == 0:
                break
            self.sh_name += chr(c)
```

Something like that. But that goes beyond the scope of this PR, so I'm ok with your latest commit and we clarify this later.